frb_example/pure_dart/rust/src/api.rs

```diff
@@ -599,10 +599,13 @@ pub struct Event {
 }
 
 pub fn register_event_listener(listener: StreamSink<Event>) -> Result<()> {
-    if let Ok(mut guard) = EVENTS.lock() {
-        *guard = Some(listener);
+    match EVENTS.lock() {
+        Ok(mut guard) => {
+            *guard = Some(listener);
+            Ok(())
+        }
+        Err(err) => Err(anyhow!("Could not register event listener: {}", err)),
```

Comment:
Btw I am not sure what will happen if a stream function returns a Result::Err, maybe it will not be propagated correctly to rust. Maybe we should add a separate test (probably in a separate PR) to confirm the behavior.

Reply:
I expect that the handler will correctly insert an error event, which should be thrown on the Dart side. But otherwise I agree we need more testing on this.

Reply:
IIRC it may not handle this case... That should be a bug though - so we may test it or raise an issue about it

Reply:
Raised #637 to address this issue.